pkg/scheduler/internal/queue/scheduling_queue.go

```diff
@@ -1141,6 +1161,11 @@ func (p *PriorityQueue) requeuePodViaQueueingHint(logger klog.Logger, pInfo *fra
 
 // NOTE: this function assumes lock has been acquired in caller
 func (p *PriorityQueue) movePodsToActiveOrBackoffQueue(logger klog.Logger, podInfoList []*framework.QueuedPodInfo, event framework.ClusterEvent, oldObj, newObj interface{}) {
+	if !p.isInterestedEvent(logger, event) {
```

Comment:
We can also add this check for  `moveAllToActiveOrBackoffQueue` to save one iteration for unschedulablePods?
https://github.com/kubernetes/kubernetes/blob/60dcf7fd8dbbb05fa02a08217001f98c5b1507c2/pkg/scheduler/internal/queue/scheduling_queue.go#L1081